terraform/deployments/ecr/main.tf

```diff
@@ -209,6 +209,42 @@ resource "aws_iam_user_policy" "github_ecr_user_policy" {
   })
 }
 
+resource "aws_ecr_lifecycle_policy" "ecr_lifecycle_policy" {
+  for_each   = toset([for repo in local.repositories : aws_ecr_repository.repositories[repo].name])
+  repository = each.key
+
+  policy = <<EOF
+  {
+    "rules": [
+        {
+            "rulePriority": 1,
+            "description": "Expire images older than 30 days",
+            "selection": {
+                "tagStatus": "untagged",
+                "countType": "sinceImagePushed",
+                "countUnit": "days",
+                "countNumber": 30
+            },
+            "action": {
+                "type": "expire"
+            },
+            "rulePriority": 1,
+            "description": "Keep last 20 images",
```

Comment:
We might need to update the deploy script to not re-build an image if it already exists. Currently, if someone were to manually deploy (i.e. which builds and pushes a new image) 20 times, we could potentially lose the version we have deployed in production.

Reply:
thats a good point, would it be easier to have another rule with higher priority, that always leave one image tagged 'latest' ? 
something like:
`{
            "rulePriority": 1,
            "description": "Rule 1",
            "selection": {
                "tagStatus": "tagged",
                "tagPrefixList": ["latest"],
                "countType": "imageCountMoreThan",
                "countNumber": 1
            },
            "action": {
                "type": "expire"
            }`

Reply:
Hmm, I don't think we can use "latest" tho - as that's just assigned to whatever image was publish last. Maybe we need to switch to better tagging scheme e.g. release_<number> that is more representative version order?

Reply:
tbh, I think this can be a further improvement and shouldn't stop these initially lifecycle rule being enabled.